### PR TITLE
[DOCS] Revises the glossary for 1.0

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
## Description
- Adds the 1.0 revised glossary

Note: This is an updated version of [PR 9714](https://github.com/great-expectations/great_expectations/pull/9714)

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]

### CODE REVIEW

File: docs/docusaurus/docs/reference/learn/glossary.md [hunk 1]
```diff
@@ -3,6 +3,25 @@ id: glossary
 title: "Glossary"
 ---
 
-import InProgress from '../../core/_core_components/_in_progress.md'
+[Action](/core/trigger_actions_based_on_results/trigger_actions_based_on_results.md): A component that integrates Great Expectations with other tools based on a Validation Result, like sending notifications based on the validation's outcomes.
```
Comment (billdirks) [inline docs/docusaurus/docs/reference/learn/glossary.md]: We could say that this is configurable on a Checkpoint, or maybe that is best left in the linked document.

File: docs/docusaurus/docs/reference/learn/glossary.md [hunk 2]
```diff
@@ -3,6 +3,25 @@ id: glossary
...
-<InProgress/>
+[Batch Definition](/core/connect_to_data/connect_to_data.md): A configuration for how a Data Asset should be divided into individual batches for testing.
+
+[Checkpoint](/core/trigger_actions_based_on_results/create_a_checkpoint_with_actions.md): An object that groups Validation Definitions and runs them with shared parameters and automated Actions.  Checkpoints are the primary means for validating data in a production deployment of Great Expectations.
+
+[Data Asset](/core/connect_to_data/connect_to_data.md): A collection of records within a Data Source which are usually grouped based on the underlying data system and can be further sliced into Batches according to a desired specification.
+
+[Data Context](/core/set_up_a_gx_environment/create_a_data_context.md): A Data Context stores all of the configuration for a project in Great Expectations, and provides an API to access and update it.
+
+[Data Docs](/core/configure_project_settings/configure_data_docs/configure_data_docs.md): Human readable documentation generated from Great Expectations metadata detailing Expectations, Validation Results, etc.
```
Comment (billdirks) [inline docs/docusaurus/docs/reference/learn/glossary.md]: These are a static web site vs human readable strings/text.

File: docs/docusaurus/docs/reference/learn/glossary.md [hunk 3]
```diff
@@ -3,6 +3,25 @@ id: glossary
...
+
+[Data Source](/core/connect_to_data/connect_to_data.md): An object that tells GX how to access a specific source of external data and provides a unified API within GX for organizing that data into Data Assets which can be retrieved.
```
Comment (billdirks) [inline docs/docusaurus/docs/reference/learn/glossary.md]: Consider s/access a specific source of external data/connect to a specific source of external data/.

File: docs/docusaurus/docs/reference/learn/glossary.md [hunk 4]
```diff
@@ -3,6 +3,25 @@ id: glossary
...
+
+[Data Docs](/core/configure_project_settings/configure_data_docs/configure_data_docs.md): Human readable documentation generated from Great Expectations metadata detailing Expectations, Validation Results, etc.
+
+[Data Source](/core/connect_to_data/connect_to_data.md): An object that tells GX how to access a specific source of external data and provides a unified API within GX for organizing that data into Data Assets which can be retrieved. 
+
+[Expectation](/core/define_expectations/create_an_expectation.md): A verifiable assertion about data.
+
+[Expectation Suite](/core/define_expectations/organize_expectation_suites.md): A group of Expectations that describe how data should be tested.
+
+[Validation Definition](/core/run_validations/create_a_validation_definition.md): A fixed reference that links an Expectation Suite to data that it describes.
```
Comment (billdirks) [inline docs/docusaurus/docs/reference/learn/glossary.md]: I would remove `fixed`. We discussed making these immutable but we didn't end up enforcing/implementing that.

File: docs/docusaurus/docs/reference/learn/glossary.md [hunk 5]
```diff
@@ -3,6 +3,25 @@ id: glossary
...
+
+<!-- Update this link to point to the "Review a Validation Result object" guide when it is complete. -->
+[Validation Result](/core/run_validations/run_a_validation_definition.md): A report generated when data is validated against an Expectation or Expectation Suite.
```
Comment (billdirks) [inline docs/docusaurus/docs/reference/learn/glossary.md]: I'd s/report/object/. You can also add that we provide facilities to view this as a json object.

**Review by billdirks — COMMENTED** [review]
I've added a few comments throughout. The only one I'd say you should definitely take is removing "fixed" from the Validation Definition definition.

Another concept you should add is `Batch`. The user only gets their hands on this in an interactive/exploratory workflow but it is a representation of a group of data that the user can interactive run validations on. Otherwise, `Batch`es do exist internal to checkpoints and running validations that way.

File: docs/docusaurus/docs/reference/learn/glossary.md (hunk 5 above)
Comment (Rachel-Reverie) [reply docs/docusaurus/docs/reference/learn/glossary.md]: Changed to object, didn't add the additional content about how to view it -- seems like there's lots we could put there if we started expanding, better to leave it in the linked content I think.

**Review by billdirks — APPROVED** [review]
This looks great, thank you!

**Pull request merged**